chromium_src/components/payments/content/payment_request.cc

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/payments/content/payment_request.h"
+
+#include "base/bind.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "brave/components/payments/content/buildflags/buildflags.h"
+#include "content/public/browser/browser_context.h"
+#include "third_party/blink/public/mojom/payments/payment_request.mojom.h"
+
+using payments::mojom::PaymentErrorReason;
+
+#if BUILDFLAG(ENABLE_PAY_WITH_BAT)
+namespace payments {
+
+void PaymentRequest::OnError(PaymentErrorReason reason,
+                             std::string err) {
+  client_->OnError(reason, err);
+}
+
+void PaymentRequest::Init(
+    mojo::PendingRemote<mojom::PaymentRequestClient> client,
+    std::vector<mojom::PaymentMethodDataPtr> method_data,
+    mojom::PaymentDetailsPtr details,
+    mojom::PaymentOptionsPtr options) {
+  bool result = false;
+  content::WebContents* contents = web_contents();
+  content::BrowserContext* context = contents->GetBrowserContext();
+
+  for (size_t i = 0; i < method_data.size(); i++) {
+    if (method_data[i]->supported_method == brave_rewards::kBatPaymentMethod) {
+      result = true;
+      break;
+    }
+  }
+
+  if (!result) {
+    Init_ChromiumImpl(std::move(client), std::move(method_data),
+                      std::move(details), std::move(options));
+    return;
+  }
+
+  // For BAT payment method, every item should have an associated SKU token
+  if (details->display_items.has_value()) {
+    for (const mojom::PaymentItemPtr& display_item : *details->display_items) {
+      if (!(display_item->sku.has_value())) {
+        log_.Error(brave_rewards::errors::kInvalidData);
+        TerminateConnection();
+        return;
+      }
+    }
+  } else {
+    log_.Error(brave_rewards::errors::kInvalidData);
+    TerminateConnection();
+    return;
+  }
+
+  // BAT payment method only works in regular mode.
+  if (context->IsOffTheRecord()) {
```

Comment:
why is PaymentRequest checking to see if this is OTR? This really doesn't seem like the right place for any of this stuff

Reply:
also you should be using delegate_->IsOffTheRecord

Reply:
but you should be checking that in bat payment specific code, not here